numpy/core/src/multiarray/methods.c

```diff
@@ -1229,18 +1229,67 @@ array_sort(PyArrayObject *self, PyObject *args, PyObject *kwds)
 {
     int axis=-1;
     int val;
-    NPY_SORTKIND sortkind = NPY_QUICKSORT;
+    NPY_SORTKIND sortkind = -1;
     PyObject *order = NULL;
+    PyObject *by = NULL;
     PyArray_Descr *saved = NULL;
     PyArray_Descr *newd;
-    static char *kwlist[] = {"axis", "kind", "order", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&O:sort", kwlist,
+
+    static char *kwlist[] = {"axis", "kind", "order", "by", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|iO&OO:sort", kwlist,
                                     &axis,
                                     PyArray_SortkindConverter, &sortkind,
-                                    &order)) {
+                                    &order, &by)) {
         return NULL;
     }
+    // If keys are None and We have complex types use c.real, c.imag as keys for lexsort
+    if (by == Py_None || by == NULL) {
+        if (PyArray_ISCOMPLEX(self)) {
+            PyArrayObject *real = (PyArrayObject*)PyObject_GetAttrString(
+                (PyObject*)self, "real");
+            if (real == NULL) {
+                return NULL;
+            }
+            PyArrayObject *imag = (PyArrayObject*) PyObject_GetAttrString(
+                    (PyObject*)self, "imag");
+            if (imag == NULL) {
+                Py_DECREF(real);
+                return NULL;
+            }
+            by = PyTuple_Pack(2, imag, real);
+            Py_DECREF(real);
+            Py_DECREF(imag);
+            if (by == NULL) {
+                return NULL;
+            }
+        }
+    }
+    if (by != Py_None && by != NULL) {
+        if (sortkind == -1) {
+            sortkind = NPY_STABLESORT;
+        }
+        if (sortkind != NPY_STABLESORT) {
+            PyErr_SetString(PyExc_ValueError, "When using by argument " \
+                            "sort-kind can be only \'stable\'.");   
+            return NULL;
+        }
+        if (!PyTuple_CheckExact(by) && !PyList_CheckExact(by)) {
+            by = PyTuple_Pack(1, by);
+            if (by == NULL) {
+                return NULL;
+            }
+        }
+        PyObject *ret = PyArray_KeySort(self, by, axis);
+        if (ret == NULL){
+            return NULL;
+        }
+        Py_DECREF(ret);
+        Py_RETURN_NONE;
```

Comment:
```suggestion
       return PyArray_KeySort(self, by, axis);
```
since `PyArray_KeySort` already returns `None`